p2p/protocol/identify/id.go

```diff
@@ -592,9 +592,10 @@ func diff(a, b []string) (added, removed []string) {
 func (ids *idService) consumeMessage(mes *pb.Identify, c network.Conn, isPush bool) {
 	p := c.RemotePeer()
 
+	// TODO (sukunrt): Fix this
 	supported, _ := ids.Host.Peerstore().GetProtocols(p)
-	added, removed := diff(supported, mes.Protocols)
-	ids.Host.Peerstore().SetProtocols(p, mes.Protocols...)
+	added, removed := diff(protocol.ConvertToStrings(supported), mes.Protocols)
+	ids.Host.Peerstore().SetProtocols(p, protocol.ConvertFromStrings(mes.Protocols)...)
```

Comment:
Probably makes the most sense to make `diff` take `protocol.ID`s instead of `string`s. Since we're getting the list of strings from the Protobuf, we don't really have a choice but to convert from strings once.

Reply:
Changed the diff function. It looks much better now.